DiscordBridge v1.8.0

DiamondIceNS opened the pull request:
I've cleaned everything up and supplied verbose Javadoc for your reviewing pleasure.

Once you accept the pull request, I'll remove my repo and re-clone from gg.obsidian.

jacobwgillespie committed:
v1.4.0

- adds ability to reload the plugin configuration and reconnect to
Discord (fixes #15)
- automatically reconnects to Discord on disconnect (fixes #9)
- strips colors from messages when sending to Discord (fixes #4)
- optionally sends death messages to Discord (fixes #12)
- adds option to use display name in message templates (fixes #13)
- adds option to use world name in message templates (fixes #14)
- changes backing API library to fix changes on Discord's end (fixes
#18)

Pawned committed:
Update plugin.yml

Prevent conflicts with Spacebukkit.

jacobwgillespie committed:
Merge pull request #23 from Pawned/patch-1

Update plugin.yml

jacobwgillespie committed:
v1.4.1

jacobwgillespie committed:
v1.4.2

- update Discord API library

jacobwgillespie committed:
v1.4.3

- disable voice support (to fix library issue on shared hosting)

jacobwgillespie committed:
Switch to gradle for builds

jacobwgillespie committed:
Build shadowJar after success

jacobwgillespie committed:
Disable voice on reload as well

jacobwgillespie committed:
v1.4.4

- keep voice disabled on reload
- build project with gradle

jacobwgillespie committed:
v1.4.5

- do not translate colors from Discord

jacobwgillespie committed:
v1.5.0

- allow disabling of main chat
- allow customization of messages if the player is vanished

jacobwgillespie committed:
v1.6.0

- Upgrade Discord API library to 1.3.0_188

jacobwgillespie committed:
Update gradle indentation

jacobwgillespie committed:
Update JDA, add ability to specify bot token

jacobwgillespie committed:
Add README update

jacobwgillespie committed:
v1.7.0

jacobwgillespie committed:
v1.7.1

Fix error where plugin crashes without bot token

Commit:
General
- Added ability to register Discord with Minecraft

Minecraft
- Added getDiscordIds command
- Added registeralias command
- Added config file to store aliases

Discord
- Added listserver command
- Added Oikos

Commit:
General
- Added ability to register Discord with Minecraft

Minecraft
- Added getDiscordIds command
- Added registeralias command
- Added config file to store aliases

Discord
- Added listserver command
- Added Oikos

MadMooAU committed:
Fixed part 2 of Oikos

MadMooAU committed:
Fixed part 2 of Oikos

DiamondIceNS committed:
Merge pull request #1 from MadMooAU/master

Fixed part 2 of Oikos

Commit:
Updated to JDA 3.x
Updated license

Commit:
Updated to JDA 3.x
Updated license

Commit:
Removed hardcoded id to make message generic
Fixed oikos emoji(?)
Added commands to plugins.yml

Commit:
Removed hardcoded id to make message generic
Fixed oikos emoji(?)
Added commands to plugins.yml

Commit:
Marina actually works now
Fixed relay function
Alias feature now works properly
Broke private messaging responses
Oikos Part 2 is still broken

Commit:
Marina actually works now
Fixed relay function
Alias feature now works properly
Broke private messaging responses
Oikos Part 2 is still broken

Commit:
Fixed private messaging
Fixed Oikos Part 2 for reals

Commit:
Fixed private messaging
Fixed Oikos Part 2 for reals

Commit:
1.7.3
Restructured commands
Added /discord get online

Commit:
1.7.3
Restructured commands
Added /discord get online

Commit:
Removed commands from plugin.kt

Commit:
Removed commands from plugin.kt

Commit:
Various fixes

Commit:
Various fixes

Commit:
whoops

Commit:
whoops

Commit:
added handler for Marina command

Commit:
added handler for Marina command

Commit:
Re-added the Util class after nuking it from orbit because I'm a dumbass

Commit:
Re-added the Util class after nuking it from orbit because I'm a dumbass

Commit:
Rearranged a lot of stuff

Commit:
Rearranged a lot of stuff

Commit:
1.7.5
Added scripted responses
Unified the message sending methods under Plugin
Broke up the formatMessage method into specific cases

Commit:
1.7.5
Added scripted responses
Unified the message sending methods under Plugin
Broke up the formatMessage method into specific cases

Commit:
1.7.6
Added F
Added 8ball
Added Insult
Added Rate
Restructured a lot of the code to embrace more code resuse

Commit:
1.7.6
Added F
Added 8ball
Added Insult
Added Rate
Restructured a lot of the code to embrace more code resuse

Commit:
partway there to having everything config-driven

Commit:
partway there to having everything config-driven

Commit:
Implemented most of a reflection-based command handling system

Commit:
Implemented most of a reflection-based command handling system

Commit:
Unfinished transition to a completely reflection-based system

Commit:
Unfinished transition to a completely reflection-based system

Commit:
It'll compile, but is it broken? Probably.

Commit:
It'll compile, but is it broken? Probably.

Commit:
fixed no commands; some commands are still spotty

Commit:
fixed no commands; some commands are still spotty

Commit:
Build is now mostly stable on the new system

Commit:
Build is now mostly stable on the new system

Commit:
Minor changes, added todo list

Commit:
Minor changes, added todo list

Commit:
- Reverted package name, license, version, etc back to gg.obsidian to
prepare for pull request

- Added basic Dynmap integration

- Fixed text issues with Parboiled

- Began adding verbose Javadoc

Commit:
- Reverted package name, license, version, etc back to gg.obsidian to
prepare for pull request

- Added basic Dynmap integration

- Fixed text issues with Parboiled

- Began adding verbose Javadoc

Commit:
1.8.0

- Finished Javadoc

- Cleaned up some stuff

- Updated README.md

Commit:
1.8.0

- Finished Javadoc

- Cleaned up some stuff

- Updated README.md

jacobwgillespie commented:
This looks awesome!  😄  I rebased this against the latest `master` so that it is mergeable and pushed it up as the [`next-major-version`](https://github.com/the-obsidian/DiscordBridge/tree/next-major-version) branch.  Can you [recreate this PR](https://github.com/the-obsidian/DiscordBridge/compare/next-major-version?expand=1) based on that branch?  I'll give it one final pass, but at the moment my only other comment is:

Can we change this to version `2.0.0`?  Since the config file format is changing and isn't backwards-compatible, we should bump the major version rather than the minor.

👍

DiamondIceNS commented:
Forgive my naïvite but what do you mean by "recreate this PR"?

Also 2.0.0 is fine by me. Your project, your org, your rules. 👍

On Apr 28, 2017 8:47 PM, "Jacob Gillespie" <notifications@github.com> wrote:

> This looks awesome! 😄 I rebased this against the latest master so that
> it is mergeable and pushed it up as the next-major-version
> <https://github.com/the-obsidian/DiscordBridge/tree/next-major-version>
> branch. Can you recreate this PR
> <https://github.com/the-obsidian/DiscordBridge/compare/next-major-version?expand=1>
> based on that branch? I'll give it one final pass, but at the moment my
> only other comment is:
>
> Can we change this to version 2.0.0? Since the config file format is
> changing and isn't backwards-compatible, we should bump the major version
> rather than the minor.
>
> 👍
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/the-obsidian/DiscordBridge/pull/45#issuecomment-298139574>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKXoN33iBnp9y4BhPAQQjYRUkQoLO6sdks5r0paWgaJpZM4NMGJB>
> .
>

jacobwgillespie commented:
> Forgive my naïvite but what do you mean by "recreate this PR"?

No worries!  Just visit that link and create a new PR - it'll be requesting that `next-major-version` be merged into `master`.  That'll clear up the merge conflicts, otherwise your existing commits are all there.  You can close this PR since the new one'll replace it.

> Also 2.0.0 is fine by me. Your project, your org, your rules.

Cool, just going off of semver (http://semver.org/) where any backwards-incompatible change bumps the major version.  Also definitely feel free to "take over" a bit too, since it's now *our* project.  :)

Commit:
2.0.0

Bumped up a major version number to better comply with version system standards

DiamondIceNS committed:
Merge branch 'next-major-version' into master
